projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/DeploymentService.java

```diff
@@ -108,7 +108,21 @@ private void validateFieldsCanBePatched(
           "job_version",
           "same as current job version when using PATCH request.",
           mergedDeployment.getGitCommitSha(),
-          "Use PUT HTTP request to change job version.");
+          "Use POST HTTP request to change job version.");
+    } else if (mergedDeployment.getPythonVersion() != null
+        && !mergedDeployment.getPythonVersion().equals(oldDeployment.getPythonVersion())) {
+      throw new ApiConstraintError(
```

Comment:
What does the response look like in the case of this error? I think it will be valuable if we incorporate both old and new values instead of only new one.

Reply:
Refactored to better present the old and new versions. Thank you for the review